modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -80,12 +91,36 @@ public ArtifactPluginDescriptor create(File pluginFolder) throws ArtifactDescrip
     }
 
     descriptor.setClassLoaderModel(classLoaderModelBuilder.build());
-
+    descriptor.setExtensionModelDescriptorProperty(lookupJsonDescriptor(pluginFolder));
     descriptor.setBundleDescriptor(createDefaultPluginBundleDescriptor(descriptor.getName()));
 
     return descriptor;
   }
 
+  private LoaderDescriber lookupJsonDescriptor(File pluginFolder) {
+    LoaderDescriber loaderDescriber = null;
+    final File mulePluginJsonFile = new File(pluginFolder, META_INF + separator + MULE_PLUGIN_JSON);
+    if (mulePluginJsonFile.exists()) {
+
+      MulePluginModel mulePluginModel = getMulePluginJsonDescriber(mulePluginJsonFile);
+      Optional<MulePluginLoaderProperty> extensionModelDescriptor = mulePluginModel.getExtensionModelLoaderDescriptor();
```

Comment:
Is is as confusing as before, you are asking for a "loaderDescriptor" which returns a "MulePluginLoaderProperty". MulePluginLoaderProperty should be <Something>LoaderDescriptor